tests/test_composable.py

```diff
@@ -67,11 +65,18 @@ def test_composable_valid(
     with open(current_config_path) as f:
         current_config = f.read()
 
-    # expecting output config has `use xxx from yyy`
-    config_name = test_config_file.split("/")[-1].removesuffix(".c4m")
-    config_path = "/".join((CONFIGS / test_config_file).__str__().split("/")[:-1])
-    use_output = f'use {config_name} from "{config_path}"'
-    assert use_output in current_config
+    if replace:
+        # replaces current config with content of incoming
+        with open(CONFIGS / test_config_file) as f:
+            original_config = f.read()
+            assert current_config == original_config
+    else:
+        # adds incoming as component
+        # expecting output config has `use xxx from yyy`
+        config_name = test_config_file.split("/")[-1].removesuffix(".c4m")
```

Comment:
```python
>>> pathlib.Path('foo/bar/baz/test.c4m').stem
'test'
```

lets use `.stem` instead

https://docs.python.org/3.12/library/pathlib.html#pathlib.PurePath.stem

```suggestion
        config_name = test_config_file.stem
```